vm/systemSmartContracts/delegation.go

```diff
@@ -872,6 +873,20 @@ func (d *delegation) addValueToFund(key []byte, value *big.Int) error {
 	return err
 }
 
+func (d *delegation) addDelegatorIfNew(dStatus *DelegationContractStatus, address []byte) {
```

Comment:
I think this can be optimized to something like:
```GO
func (d *delegation) addDelegatorIfNew(dStatus *DelegationContractStatus, address []byte) {
	for i := range dStatus.Delegators {
		if bytes.Equal(address, dStatus.Delegators[i]) {
			return
		}
	}

	dStatus.Delegators = append(dStatus.Delegators, address)
}
```

Reply:
no need for for here. _, dData, err := d.getOrCreateDelegatorData(args.CallerAddr) the first return parameter is called isNew :). plus this function is called only if it was new. No need for these checks.

Reply:
the simple append is good enough, while checking the isNew return param.

Reply:
Done.

Reply:
addDelegatorIfNew(dStatus *DelegationContractStatus, address []byte). This is called after  d.getOrCreateDelegatorData(address), thus you do not need to get it again. 
 addDelegatorIfNew function is not needed. in the delegate function you need if isNew -> append